src/github/api.rs

```diff
@@ -252,29 +252,46 @@ impl Session for GithubSession {
         state: Option<&str>,
         head: Option<&str>,
     ) -> Result<Vec<PullRequest>> {
-        let pull_requests: Result<Vec<PullRequest>> = self.client
-            .get::<Vec<PullRequest>>(
-                &format!("repos/{}/{}/pulls?state={}&head={}", owner, repo, state.unwrap_or(""), head.unwrap_or("")),
-            )
-            .map_err(|e| format!("Error looking up PRs: {}/{}: {}", owner, repo, e).into())
-            .map(|prs| {
-                prs.into_iter()
-                    .filter(|p| if let Some(head) = head {
-                        p.head.ref_name == head || p.head.sha == head
-                    } else {
-                        true
-                    })
-                    .collect::<Vec<_>>()
-            });
-        let mut pull_requests = pull_requests?;
+        let mut pull_requests = vec![];
+        let mut page = 1;
+
+        loop {
+            let mut next_prs: Vec<PullRequest> = match self.client
+                .get::<Vec<PullRequest>>(&format!(
+                    "repos/{}/{}/pulls?state={}&head={}&per_page=100&page={}",
```

Comment:
BTW, does 100/page works, or should it be 30?

Reply:
100 is the max per-page results github api will return. Grabbing more at a time means fewer api calls in total.